src/DataProvider/OperationDataProvider.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataProvider;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Exception\ItemNotFoundException;
+use ApiPlatform\Core\Identifier\Normalizer\ChainIdentifierDenormalizer;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class OperationDataProvider
```

Comment:
Shouldn't we introduce an interface?

Reply:
We may but I'm also to mark this as `@internal` unless we can find a global use for it? For now it'll be only used by the IriConverter and I really doubt it'll be used for anything else :p. 
I just want to avoid injecting everything the OperationDataProviderTrait depends on in the IriConverter.